### PR TITLE
Allow users to edit events

### PR DESCRIPTION
**Pull request by nmsulliv**
This PR allows users to edit events they've created. The doGet function checks if the event belongs to the user trying to access the link, and then populates the fields in the JSP form with the information on datastore. The doPost updates the information in datastore.

I added two functions for formatting to Utils.java because they are used in both the EventServlet and EditEventServlet. I also updated the tests for EventServlet and UserServlet. I plan on implementing the tests in the next PR.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/EditEventServlet.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    PreparedQuery queried = datastore.prepare(query);
+
+    // Checks if the event is in the list of events created by the user.
+    for (Entity e : queried.asIterable()) {
+      String userEventKey = e.getProperty("eventKey").toString();
+
+      if (userEventKey.equals(eventKeyString)) {
+        exists = true;
+      }
+    }
```
Comment (0tpyrk) [inline src/main/java/com/google/sps/servlets/EditEventServlet.java:239]: Would it be more efficient if we queried for the event we're trying to edit using its key and then checked if it's creator is the current user?

Comment (fang-helen) [reply src/main/java/com/google/sps/servlets/EditEventServlet.java:239]: I agree!

Comment (nmsulliv) [reply src/main/java/com/google/sps/servlets/EditEventServlet.java:239]: Yes it definitely would! Thanks for pointing that out.

File: src/main/java/com/google/sps/servlets/EditEventServlet.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+              entity.setProperty("firebaseID", userID);
+              datastore.put(entity);
+              request
+                  .getRequestDispatcher("/WEB-INF/jsp/access-denied.jsp")
+                  .forward(request, response);
+            }
+          }
+        }
+      }
+    } catch (IllegalArgumentException | IOException | NullPointerException | ServletException e) {
```
Comment (tc466) [inline src/main/java/com/google/sps/servlets/EditEventServlet.java:111]: It is not immediately clear to the reader which parts of the code throw these exceptions. Maybe handle them where they are thrown (like the EntityNotFoundException above), or add some comments to document where these exceptions originate.

File: src/main/java/com/google/sps/Utils.java [hunk 3]
```diff
@@ -161,4 +165,44 @@ public int compare(Entity a, Entity b) {
         };
     return result;
   }
+
+  /** Format time to standard format. */
+  public static String formatTime(String time) {
```
[the remark this answers is not in the document]
Comment (Yunni) [reply src/main/java/com/google/sps/Utils.java:170]: +1

File: src/main/java/com/google/sps/servlets/EditEventServlet.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    response.sendRedirect("/my-events.html");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    Key keyRequested;
+    try {
+      keyRequested = getEventKey(request, "Event");
+      if (keyRequested != null) {
```
Comment (Yunni) [inline src/main/java/com/google/sps/servlets/EditEventServlet.java:71]: Try to handle the corner case first to make the error handling more explicit and the code less nested.
```
if (keyRequested == null) {
  // error handling
}
// Normal processing
```

Comment (nmsulliv) [reply src/main/java/com/google/sps/servlets/EditEventServlet.java:71]: Will do, thanks!

File: src/main/java/com/google/sps/servlets/EditEventServlet.java (hunk 2 above)
Comment (nmsulliv) [reply src/main/java/com/google/sps/servlets/EditEventServlet.java:111]: Will do!

**Pull request merged**